server/services/flood.js

```diff
@@ -21,11 +21,25 @@ module.exports = {
   },
   // get cached outlook object
   get outlook () {
-    return outlook
+    try {
+      return outlook
+    } catch (err) {
+      console.error(`Get Outlook data error - [${err}]`)
+      return { dataError: true }
+    }
   },
   // set cached outlook object
   set outlook (data) {
-    outlook = data && new Outlook(data)
+    try {
+      const newData = new Outlook(data)
```

Comment:
@JFSCH I think this needs an additional `(newData.dataError)` check sooner in here.  Because the getOutlook function sets the dataError var if it fails or errors for whatever reason. But that object is still passed through to the outlook model in that situation. Which then throws a forEach error as it tries to process an invalid object.

Reply:
@teddmason Added in check as requested. Had to refactor routes/location.js and unit test coverage for code analysis check